packages/azure-kusto-ingest/src/fileDescriptor.browser.ts

```diff
@@ -22,10 +24,25 @@ export class FileDescriptor extends AbstractDescriptor implements FileDescriptor
         this.compressionType = compressionType;
         this.size = size || file.size;
         this.zipped = compressionType !== CompressionType.None || this.extension === ".gz" || this.extension === ".zip";
+        this.shouldNotCompress =
```

Comment:
You override the users' shouldNotCompress, if you get this from the user - accept his choice

Reply:
actually you don't need this 
extension will determine the format and you deal with it in prepare()

Reply:
This is same in other descriptors

Reply:
True, but this is not handling the same scenario - we want the compression decision both when the file suffix is detected (prepare) and also when the format is explicitly provided

Reply:
Sorry- i thought we are deducing the format from the extension 
Anyway- you still didnt address my first concern (users input)

Reply:
unless im missing something, we only get the extension, not an actual shouldNotCompress from the user, arent we?